docs/authorization-introduction.md

```diff
@@ -0,0 +1,124 @@
+# Authorization Introduction
+
+This document provides a general overview on what are the terms used in the Udaru authorization system and describes the structure and functionality of the authorization application.
+
+The Udaru authorization system has two main functional components:
+- Using the PBAC authorization mechanism for internal use to authorize the system internal parts against themselves,
+- Provide PBAC authorization service for external applications.
+
+## Authorization terms
+
+The access control management evolved from the need to mitigate the risk of unauthorized access to data.
+
+Several control models were built during time to approach the various access control needs: Access Control Lists, Role Based Access Control, Attribute Based Access Control, Policy Based Access Control, Risk Adaptive Access Control. A summary of these models is made on [Survey on access control models][], the document offers a description of the PBAC model that Udaru uses.
+
+Udaru is based on the [node-pbac][] module, module that as the developer claims is inspired and mostly compatible with the [AWS IAM][] policy engine.
+
+Altough the [AWS IAM Policy][] is not the documentation for [node-pbac][] it can be used as reference for details regarding policies and policy elements.
+
+## The Authorization model
+
+An access control model can be built using the **Organization**, **Team** and **User** entities. The Organizations can have Teams or Users attached to them. Teams can have sub Teams or Users attached to them. Only the Team elements can have Team elements (sub Teams) attached to it.
+
+Policies can be attached to Organizations, Teams and Users. The policy engine evaluates all policies from all levels when establishing if a User can perform an Action over a Resource.
```

Comment:
Policies **cannot** be attached to Organizations yet. The db structure is there but no endpoint or function to perform that operation.